pkg/operator/controller/externaldns/pod.go

```diff
@@ -92,6 +92,14 @@ type externalDNSContainerBuilder struct {
 func (b *externalDNSContainerBuilder) build(zone string) (*corev1.Container, error) {
 	seq := b.counter
 	b.counter++
+	// check each zone for the keyword 'privatednszones', if the provider is Azure
+	// this ensures that the provider 'azure-private-dns' is passed to the container
+	// to set the operand provider correctly
+	if b.provider == externalDNSProviderTypeAzure {
+		if strings.Contains(strings.ToLower(zone), azurePrivateDNSZonesResourceSubStr) {
+			b.provider = externalDNSProviderTypeAzurePrivate
+		}
+	}
```

Comment:
Sorry, I didn't precise the proposal I made. What I meant by moving this part in the container builder is that in the container builder we have methods for each individual provider. And this Azure specific code would feel like at home in `fillAzureFields()`, unlike here in `build()` where the code is platform agnostic.
To make this happen I think that `--provider` flag would have to be moved to each provider.

Reply:
Ok got it - this makes sense

Reply:
Ok makes total sens - I have moved the code to fillAzureFields() now